### PR TITLE
217, 256, 268 Add xml, json backends, support for tabular(sep: ";")

### PR DESCRIPTION
**Pull request by adp-atea**

### CODE REVIEW

File: notes/datasets/xml.sh [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+poetry run spinta show $BASEDIR/manifest.csv
+
+# Run server with external mode (read from db.sqlite directly)
+test -n "$PID" && kill $PID
+poetry run spinta run --mode external $BASEDIR/manifest.csv &>> $BASEDIR/spinta.log &; PID=$!
+tail -50 $BASEDIR/spinta.log
+
+# notes/spinta/client.sh    Configure client
+
+http GET "$SERVER/$DATASET/SeimoKadencija"
```
Comment (sirex) [inline notes/datasets/xml.sh:63]: I get following error, when trying to read XML:

```python
Traceback (most recent call last):
  File "spinta/api.py", line 95, in homepage
    return await create_http_response(context, params, request)
  File "spinta/utils/response.py", line 153, in create_http_response
    return await commands.getall(
  File "spinta/commands/read.py", line 125, in getall
    return render(context, request, model, params, rows, action=action)
  File "spinta/renderer.py", line 23, in render
    return commands.render(
  File "multipledispatch/dispatcher.py", line 278, in __call__
    return func(*args, **kwargs)
  File "spinta/formats/json/commands.py", line 25, in render
    return _render(
  File "spinta/formats/json/commands.py", line 52, in _render
    aiter(peek_and_stream(fmt(data))),
          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "spinta/utils/response.py", line 223, in peek_and_stream
    peek = list(itertools.islice(stream, 2))
  File "spinta/formats/json/components.py", line 16, in __call__
    for i, row in enumerate(data):
  File "spinta/accesslog/__init__.py", line 162, in log_response
    for row in rows:
  File "spinta/commands/read.py", line 110, in <genexpr>
    rows = (
  File "spinta/datasets/backends/dataframe/commands/read.py", line 398, in getall
    yield from _dask_get_all(context, query, df, backend, model, builder)
  File "spinta/datasets/backends/dataframe/commands/read.py", line 435, in _dask_get_all
    for i, row in qry.iterrows():
  File "spinta/datasets/backends/dataframe/commands/read.py", line 298, in _get_data_xml
    return _parse_xml(f.raw, source, model_props)
  File "spinta/datasets/backends/dataframe/commands/read.py", line 138, in _parse_xml
    for action, elem in etree.iterparse(data, events=('start', 'end'), remove_blank_text=True):
  File "src/lxml/iterparse.pxi", line 210, in lxml.etree.iterparse.__next__
  File "src/lxml/iterparse.pxi", line 195, in lxml.etree.iterparse.__next__
  File "src/lxml/iterparse.pxi", line 226, in lxml.etree.iterparse._read_more_events
  File "src/lxml/parser.pxi", line 1395, in lxml.etree._FeedParser.close
  File "<string>", line 0
lxml.etree.XMLSyntaxError: no element found
```

**Review by sirex — CHANGES_REQUESTED** [review]

**Review by sirex — CHANGES_REQUESTED** [review]

**Review by sirex — APPROVED** [review]
I suggest to merge this and fix remaining issues with another pull request.

File: notes/datasets/json.sh [hunk 2]
```diff
@@ -0,0 +1,585 @@
...
+#|         ...
+#|     ],
+#|     "source": "https://www.geoportal.lt/metadata-catalog/rest/find/document?start=1&max=10&f=pjson",
+#|     "startIndex": 1,
+#|     "title": "Geoportal GeoRSS.",
+#|     "totalResults": 762,
+#|     "updated": "2023-07-09T11:35:13Z"
+#| }
+# FIXME: There is one date and it is timezone-aware, because of Z suffix, which
+#        means it is in UTC timezone. Please handle timezone-aware dates correctly.
```
[the remark this answers is not in the document]
Comment (adp-atea) [reply notes/datasets/json.sh:246]: will be fixed with separate PR

File: notes/datasets/json.sh [hunk 3]
```diff
@@ -0,0 +1,585 @@
...
+#|     for i, row in qry.iterrows():
+#|   File "spinta/datasets/backends/dataframe/commands/read.py", line 310, in _get_data_json
+#|     with pathlib.Path(url).open(encoding='utf-8-sig') as f:
+#|          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#|   File "/usr/lib/python3.11/pathlib.py", line 1044, in open
+#|     return io.open(self, mode, buffering, encoding, errors, newline)
+#|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#| FileNotFoundError: [Errno 2] No such file or directory: '{url}'
+# FIXME: Parametrization does not work, because `{url}` must be replaced by
+#        page url as described with `url` param.
```
[the remark this answers is not in the document]
Comment (adp-atea) [reply notes/datasets/json.sh:525]: will be fixed with separate PR

File: notes/datasets/json.sh [hunk 4]
```diff
@@ -0,0 +1,585 @@
...
+#|   File "src/lxml/iterparse.pxi", line 195, in lxml.etree.iterparse.__next__
+#|   File "src/lxml/iterparse.pxi", line 230, in lxml.etree.iterparse._read_more_events
+#|   File "src/lxml/parser.pxi", line 1379, in lxml.etree._FeedParser.feed
+#|   File "src/lxml/parser.pxi", line 609, in lxml.etree._ParserContext._handleParseResult
+#|   File "src/lxml/parser.pxi", line 618, in lxml.etree._ParserContext._handleParseResultDoc
+#|   File "src/lxml/parser.pxi", line 728, in lxml.etree._handleParseResult
+#|   File "src/lxml/parser.pxi", line 657, in lxml.etree._raiseParseError
+#|   File "<string>", line 1
+#| lxml.etree.XMLSyntaxError: Document is empty, line 1, column 1
+# FIXME: Probably issue with the pagination
```
[the remark this answers is not in the document]
Comment (adp-atea) [reply notes/datasets/json.sh:585]: will be fixed with separate PR

**Pull request merged**